### ISSUE TITLE
Cannot read property 'then' of null

### ISSUE THREAD

**Issue by pixelgamma** [zo0r/react-native-push-notification#117]
I've got this error:

`Cannot read property 'then' of null`

Here is my code:

```
import React, { Component } from 'react';

import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  ActionSheetIOS
} from 'react-native';

import * as Animatable from 'react-native-animatable';
import Linearicon from 'react-native-vector-icons/Linearicons';

import PushNotification from 'react-native-push-notification';

PushNotification.configure({

    // (optional) Called when Token is generated (iOS and Android)
    onRegister: function(token) {
        console.log( 'TOKEN:', token );
    },

    // (required) Called when a remote or local notification is opened or received
    onNotification: function(notification) {
        console.log( 'NOTIFICATION:', notification );
    },

    // ANDROID ONLY: (optional) GCM Sender ID.
    senderID: "YOUR GCM SENDER ID",

    // IOS ONLY (optional): default: all - Permissions to register.
    permissions: {
        alert: true,
        badge: true,
        sound: true
    },

    // Should the initial notification be popped automatically
    // default: true
    popInitialNotification: true,

    /**
      * IOS ONLY: (optional) default: true
      * - Specified if permissions will requested or not,
      * - if not, you must call PushNotificationsHandler.requestPermissions() later
      */
    requestPermissions: true,
});

export default class Chrono extends Component {

  constructor(props) {
    super(props);

    this.chrono = null;
  
    this.state = {
      isAlarmActive: false
    };
  }

  componentDidMount() {

    

  }

  render() {

    return (
      <TouchableOpacity style={styles.navBarBottomSide}
        onPress={() => this.setUpAlarm(this.state.isAlarmActive)}
      >
        <Linearicon name="timer" size={36} color="white" />
      </TouchableOpacity>
    )

  }

}
```

![simulator screen shot 29 juin 2016 19 11 25](https://cloud.githubusercontent.com/assets/1059964/16462092/02c3a9b2-3e30-11e6-8deb-16441a9a86d7.png)

When I put the PushNotification configuration in componentDidMount, here is what I've got:

```
componentDidMount() {

       PushNotification.configure({

        // (optional) Called when Token is generated (iOS and Android)
        onRegister: function(token) {
            console.log( 'TOKEN:', token );
        },

        // (required) Called when a remote or local notification is opened or received
        onNotification: function(notification) {
            console.log( 'NOTIFICATION:', notification );
        },

        // ANDROID ONLY: (optional) GCM Sender ID.
        senderID: "YOUR GCM SENDER ID",

        // IOS ONLY (optional): default: all - Permissions to register.
        permissions: {
            alert: true,
            badge: true,
            sound: true
        },

        // Should the initial notification be popped automatically
        // default: true
        popInitialNotification: true,

        /**
          * IOS ONLY: (optional) default: true
          * - Specified if permissions will requested or not,
          * - if not, you must call PushNotificationsHandler.requestPermissions() later
          */
        requestPermissions: true,
    }); 

  }
```
![simulator screen shot 29 juin 2016 19 27 00](https://cloud.githubusercontent.com/assets/1059964/16462080/f785c3aa-3e2f-11e6-8f72-aa5e410076d3.png)

**Comment by sorenburon** [comment]
I experienced the same. Seems like a react native (RN) & react-native-push-notification (RNPN) version mismatch.

Im on RN 0.24.1, and was using version RNPN v1.1.0 which caused the **Cannot read property 'then' of null** error.

I fixed it by installing version 1.0.7 of RNPN.

**Comment by luisfuertes** [comment]
Same error with RN v0.25.1 and RNPN v1.0.7
